proj/src/main/webapp/signup.js

```diff
@@ -127,10 +147,11 @@ function initializePasswordValidation(){
     const noMatch = document.getElementById("noMatch");
     const btnSignUp = document.getElementById("btnSignUp");
     const pass = document.getElementById("pass");
-    const passconf = document.getElementById("passconf");
+    const passconf = document.getElementById("passconf"); 
     /* Check for password confirmation 
-        enable button if and only if the password meets the requiremetns and match **/
-    $('#pass, #passconf').on('keyup', function(){
+        enable button if and only if the password meets the 
+        requirements and match each other**/
+    $('#pass, #passconf, #tags').on('keyup', function(){
```

Comment:
Wait, why are we checking the password on tag changes?

Reply:
all three things are checked because the enableSignUp function checks if all three satisfies the requirements before enabling the sign up button. enableSignUp uses the validity from checking if the passwords match and the presence of tags

Reply:
the `enableSignUp` function checks for matching password requirements along with the presence of tags. if any of the fields change, the signup button will disabled or enables accordingly

Reply:
Both the comment above and the name of the function on line 107 are misleading, then.  This function clearly does more than check whether the password contains a number!  I think they should be updated to reflect that.

Reply:
I think you're confused Matt. This method is called `initializePasswordValidation`, not `containsnumber`